Content.Server/GameObjects/Components/Conveyor/ConveyorSwitchComponent.cs

```diff
@@ -0,0 +1,233 @@
+                _state = value;
+
+                if (Owner.TryGetComponent(out AppearanceComponent appearance))
+                {
+                    appearance.SetData(ConveyorVisuals.State, value);
+                }
+            }
+        }
+
+        private ConveyorGroup Group => EntitySystem.Get<ConveyorSystem>().EnsureGroup(Id);
+
+        /// <summary>
+        ///     Syncs this switch with the state of another
+        /// </summary>
+        /// <param name="state">The state to sync with</param>
+        public void SyncState(ConveyorState state)
+        {
+            if (State == ConveyorState.Loose)
+            {
+                return;
+            }
+
+            State = state == ConveyorState.Loose
+                ? ConveyorState.Off
+                : state;
+        }
+
+        public void Connect(IEntity user, ConveyorComponent conveyor)
+        {
+            Group.AddConveyor(conveyor);
+
+            user.PopupMessage(user, Loc.GetString("Conveyor linked with id {0}.", Id));
+        }
+
+        /// <summary>
+        ///     Cycles this conveyor switch to its next valid state
+        /// </summary>
+        /// <returns>
+        ///     true if the switch can be operated and the state could be cycled,
+        ///     false otherwise
+        /// </returns>
+        private bool NextState()
+        {
+            if (Owner.HasComponent<ItemComponent>())
+            {
+                State = ConveyorState.Loose;
+                Group.SetState(this);
+
+                return false;
+            }
+
+            State = State switch
+            {
+                ConveyorState.Off => ConveyorState.Forward,
+                ConveyorState.Forward => ConveyorState.Reversed,
+                ConveyorState.Reversed => ConveyorState.Off,
+                ConveyorState.Loose => ConveyorState.Off,
+                _ => throw new ArgumentOutOfRangeException()
+            };
+
+            Group.SetState(this);
+
+            return true;
+        }
+
+        private bool ToolUsed(IEntity user, ToolComponent tool)
+        {
+            if (!Owner.HasComponent<ItemComponent>() &&
+                tool.UseTool(user, Owner, ToolQuality.Prying))
+            {
+                return Itemize();
+            }
+
+            return false;
+        }
+
+        private void SyncWith(IEntity user, ConveyorSwitchComponent other)
+        {
+            Id = other.Id;
+            Owner.PopupMessage(user, Loc.GetString("Switch changed to id {0}.", Id));
+        }
+
+        private bool Itemize()
+        {
+            if (Owner.HasComponent<ItemComponent>())
+            {
+                return false;
+            }
+
+            Owner.AddComponent<ItemComponent>();
+            NextState();
+            Owner.Transform.WorldPosition += (_random.NextFloat() * 0.4f - 0.2f, _random.NextFloat() * 0.4f - 0.2f);
+
+            return true;
+        }
+
+        private void DeItemize(GridCoordinates coordinates)
```

Comment:
Again, this "dynamically changing whether we're an item" stuff seems like a complete pain.